src/components/common/StarRating/StarRating.js

```diff
@@ -0,0 +1,107 @@
+import React from 'react';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faStar } from '@fortawesome/free-solid-svg-icons';
+import styles from './StarRating.modules.scss';
+import PropTypes from 'prop-types';
+import { faStar as farStar } from '@fortawesome/free-regular-svg-icons';
+
+const addMyRatingToState = (id, addRating, value, myRating, event) => {
+  event.preventDefault();
+  addRating({ id: id, myRating: value });
+};
+
+const addHoverToState = (id, addHover, value, myHover, event) => {
+  event.preventDefault();
+  addHover({ id: id, myHover: value });
+};
+
+const starsArray = [1, 2, 3, 4, 5];
+
+const StarRating = ({ myRating, stars, addRating, id, addHover, myHover }) =>
+  myRating && myRating != null ? (
```

Comment:
Nie podoba mi się ten if i to że mamy 2 razy bardzo podobny HTML ;)

Proponuję zrobić jeden HTML i każdej gwiazde nadać inną klasę w zależności:
- czy to ocena średnia 
- czy to ocena usera
- czy gwiazda jest pusta

Hover zrób po prostu za pomocą CSS, czyli na hover jakiś tam background czy color ;)

Kod odpowiedzialny za generowanie nazwy klasy najlepiej wydzielić do osobnej funkcji ;)